### PR TITLE
ARROW-6513: [CI] Rename conda requirements files to have txt extension instead of yml

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -140,12 +140,6 @@ jobs:
       - name: Fetch Submodules and Tags
         shell: bash
         run: ci/scripts/util_checkout.sh
-      - name: Show available RAM size
```
Comment (kszucs) [inline .github/workflows/python.yml:143]: Debugging snippet from 8aeec2896759e12ac385b44d61531528528dfac7

**Comment by kszucs** [comment]
@github-actions crossbow submit conda*

**Comment by kszucs** [comment]
@github-actions crossbow submit conda*

File: ci/conda_env_cpp.txt [hunk 2]
```diff
@@ -15,7 +15,8 @@
 # specific language governing permissions and limitations
 # under the License.
 
-aws-sdk-cpp
+# workaround for https://issues.apache.org/jira/browse/ARROW-13134
+aws-sdk-cpp<1.8
```
Comment (kszucs) [inline ci/conda_env_cpp.txt:19]: cc @xhochy

**Comment by xhochy** [comment]
> `github-actions crossbow submit conda*`

The conda jobs don't use these files.

**Comment by kszucs** [comment]
@github-actions crossbow submit test-conda-*

File: ci/conda_env_cpp.txt (hunk 2 above)
Comment (kszucs) [reply ci/conda_env_cpp.txt:19]: @pitrou I've pinned aws sdk in this PR. I assume the issue is still present, can we merge this?

Comment (pitrou) [reply ci/conda_env_cpp.txt:19]: It should be `aws-sdk-cpp<1.9`.

**Comment by kszucs** [comment]
@github-actions crossbow submit test-conda-*

**Review by kszucs — COMMENTED** [review]
+1, the build errors are unrelated

**Comment by ianmcook** [comment]
@kszucs It looks like this PR missed a few places where `.yml` needs to be changed to `.txt`:
- `dev/release/verify-release-candidate.bat`
- `docs/source/developers/documentation.rst`
- `docs/source/developers/python.rst`
- `docs/source/developers/cpp/windows.rst`
- `python/examples/minimal_build/build_conda.sh`
- a comment in `cpp/cmake_modules/ThirdpartyToolchain.cmake`

**Pull request closed without merging**